Source/AliveLibAO/Abe.cpp

```diff
@@ -9535,9 +9535,197 @@ void Abe::State_155_ElumJumpToFall_42E060()
+            {
+                field_110_state.door = DoorStates::eUnknown_4;
+                auto pTlv = static_cast<Path_Reset*>(gMap_507BA8.TLV_Get_At_446260(
+                    FP_GetExponent(field_A8_xpos),
+                    FP_GetExponent(field_AC_ypos),
+                    FP_GetExponent(field_A8_xpos),
+                    FP_GetExponent(field_AC_ypos),
+                    TlvTypes::ResetPath_115
+                ));
+                if (pTlv)
+                {
+                    if (pTlv->field_18_clearIds)
+                    {
+                        for (short i = pTlv->field_1A_from; pTlv->field_1A_from <= pTlv->field_1C_to; i++ )
+                        {
+                            if (i != pTlv->field_1E_exclude && i > 1)
+                            {
+                                SwitchStates_Set(i, 0);
+                            }
+                        }
+                    }
+                    if (pTlv->field_20_clearObjects)
+                    {
+                        gMap_507BA8.sub_447430(pTlv->field_22_path);
+                    }
+                }
+            }
+            return;
+        }
+        case DoorStates::eUnknown_4:
+        {
+            auto pDoorTlv = static_cast<Path_Door*>(gMap_507BA8.TLV_Get_At_446260(
+                FP_GetExponent(field_A8_xpos),
+                FP_GetExponent(field_AC_ypos),
+                FP_GetExponent(field_A8_xpos),
+                FP_GetExponent(field_AC_ypos),
+                Door_6
+            ));
+            field_F0_pTlv = pDoorTlv;
+            gMap_507BA8.field_1E_door = 1;
+            const auto changeEffect = kPathChangeEffectToInternalScreenChangeEffect_4CDC78[pDoorTlv->field_3A_wipe_effect];
+            short flag = 0;
+            if (changeEffect == CameraSwapEffects::eEffect5_1_FMV || changeEffect == CameraSwapEffects::eEffect11 )
+            {
+                flag = 1;
+            }
+            gMap_507BA8.SetActiveCam_444660(
+                pDoorTlv->field_18_level,
+                pDoorTlv->field_1A_path,
+                pDoorTlv->field_1C_camera,
+                changeEffect,
+                pDoorTlv->field_3C_movie_number,
+                flag
+            );
+            field_110_state.door = DoorStates::eUnknown_5;
+            field_196_door_id = pDoorTlv->field_24_target_door_number;
+            return;
+        }
+        case DoorStates::eUnknown_5:
+        {
+            field_B2_lvl_number = gMap_507BA8.field_0_current_level;
+            field_B0_path_number = gMap_507BA8.field_2_current_path;
+            gMap_507BA8.field_1E_door = 0;
+            auto pPathDoor = static_cast<Path_Door*>(gMap_507BA8.TLV_First_Of_Type_In_Camera_4464A0(Door_6, 0));
+            field_F0_pTlv = pPathDoor;
+
+            while (pPathDoor->field_20_door_number != field_196_door_id)
+            {
+                pPathDoor = static_cast<Path_Door*>(Path_TLV::TLV_Next_Of_Type_446500(
```

Comment:
this is wrong because you've missed checking a door 

get first (don't get check it)
get next (always checked from here on)

Reply:
actually I think that is correct due to the while - but should field_F0_pTlv be assigned even when it isn't the door we are looking for? 🤔

Reply:
I think the code's alright as is